### DISCUSSION TITLE
Why is opencv's processor detection considered broken?

### DISCUSSION THREAD

**Discussion by horenmar** [microsoft/vcpkg#20656] [Q&A] [answered]
[OpenCV's portfile overrides OpenCV's CPU detection](https://github.com/microsoft/vcpkg/blob/973a7d517c09c8cfb7e6a548fcc260ca34ba7b60/ports/opencv4/portfile.cmake#L318-L323) because it considers it broken. The [detection seems correct](https://github.com/opencv/opencv/blob/master/cmake/OpenCVDetectCXXCompiler.cmake) given the documentation of `CMAKE_SYSTEM_PROCESSOR`, so I wonder if it is because the portfile is being run in script mode, or because I misunderstand the CMake docs.

**Comment by cenit** [comment, marked as answer]
because it works only for host and does not consider target
